okhttp/src/main/java/okhttp3/internal/http/RetryAndFollowUpInterceptor.kt

```diff
@@ -59,8 +59,9 @@ class RetryAndFollowUpInterceptor(private val client: OkHttpClient) : Intercepto
     val call = realChain.call
     var followUpCount = 0
     var priorResponse: Response? = null
+    var newExchangeFinder = true
```

Comment:
Here’s a potentially consequential behavior change.

We used to retain the same `ExchangeFinder` for all follow-up calls, as long as host+port stuff didn’t change. That means that a route failure for the initial request would impact an eventually-successful redirect or authorization.

With this change we retain only the `Connection`. We don’t remember previous failed routes.

I think it’s both simpler and more robust.

(Though I’m not happy with the name ExchangeFinder, or even Exchange for that matter.)